README.md

```diff
@@ -1,18 +1,129 @@
 
-## Advantages
+*Implement effectful computation in a modular way!*
+
+The main and only monad is built upon `Eff` from `Control.Eff`. `Eff r a` is parameterized by
+the effect-list `r` and the monadic-result type `a` similar to other monads.
+
+In case you know monad transformers or mtl: This library provides only one monad that includes all your effects instead of layering different transformers. It is not necessary to lift the computations through a monad stack. Also, it is not required to lift every `Monad*` (like `MonadError`) typeclass though all transformers.
+
+## Quickstart
+
+### The effect list
+
+The effect list `r` in the type `Eff r a` is a central concept in this library.
+It is a type-level list containing effect types. If `r` is ... then,
+
+If `r` is the empty list, then the computation `Eff r` (or `Eff '[]`) does not contain any effects to be handled and therefore is a pure computation. In this case, the result value can be retrieved by `run :: Eff '[] a -> a`
+
+For programming within the `Eff r` monad, it is never necessary to list all effects that can appear.
+It suffices to state what types of effects are at least required.
+This is done via the `Member t r` typeclass. It describes that the type `t` occurs inside the list `r`.
+
+Effect-handler functions, mostly called `run*`, `exec*` or `eval*`, typically have a signature like `Eff (x ': r) a -> Eff r a`. Where `x` is the type of effect to be handled, like `Writer w` or `State s`. The transformation from the longer list of effects `(x ': r)` to just `r` is a type-level indicator that the effect `x` has been handled. Depending on the effect, some additional input might be required or some output is produced.
+
+## Most common effects
+
+The most common effects used are `Writer`, `Reader`, `Exception` and `State`.
+
+Every effect has 2 important functions
+
+* how to generate the effect.
+* how the effect is done
+
+For the basic effects, the types of those functions clearly state the intent of the effect.
+
+In this section, the utility functions of the effects are not listed. Once you understand the basic usage of an effect, the type of any utility function typically explains what is does.
+
+Of course, the best learning progress is done by experimenting inside `ghci`.
+
+### The Exception Effect
+
+The exception effect adds the possibility to exit a computation preemptively with an exception. Note that the exceptions from this library are handled by the programmer and have nothing to do with exceptions thrown inside the Haskell run-time.
+
+```haskell
+throwError :: Member (Exc e) r => e -> Eff r a
+runError :: Eff (Exc e ': r) a -> Eff r (Either e a)
+```
+
+An exception can be thrown using the `throwError` function. Its return type is `Eff r a` with an arbitrary type `a`. When handling the effect, the result-type changes to `Either e a` instead of
```

Comment:
@power-fungus use your discretion, but it might be helpful to describe that for `Exception`s the only kind of request you can make to the handler of the exception is to clarify your desire/intent to throw an exception. hence, we have only one kind of function/DSL primitive for introducing the `Exception` effect and that is `throwError`.

Reply:
@suhailshergill I am not sure if I get your suggestion. Do you suggest something like "The only request for the Exception-handler is the request to throw an exception" as further clarification?

Reply:
@power-fungus yes. this might be more suited for the section on custom effects. i'll leave it to your judgment